server/src/main/java/org/elasticsearch/gateway/MetadataStateFormat.java

```diff
@@ -95,25 +98,12 @@ private static void deleteFileIgnoreExceptions(Path stateLocation, Directory dir
     private void writeStateToFirstLocation(final T state, Path stateLocation, Directory stateDir, String tmpFileName)
         throws WriteStateException {
         try {
-            deleteFileIfExists(stateLocation, stateDir, tmpFileName);
-            try (IndexOutput out = stateDir.createOutput(tmpFileName, IOContext.DEFAULT)) {
-                CodecUtil.writeHeader(out, STATE_FILE_CODEC, CURRENT_VERSION);
-                out.writeInt(FORMAT.index());
-                try (XContentBuilder builder = newXContentBuilder(FORMAT, new IndexOutputOutputStream(out) {
-                    @Override
-                    public void close() {
-                        // this is important since some of the XContentBuilders write bytes on close.
-                        // in order to write the footer we need to prevent closing the actual index input.
-                    }
-                })) {
-                    builder.startObject();
-                    toXContent(builder, state);
-                    builder.endObject();
-                }
-                CodecUtil.writeFooter(out);
+            try {
+                doWriteToFirstLocation(state, stateDir, tmpFileName);
+            } catch (FileAlreadyExistsException fae) {
```

Comment:
It doesn't look like anything in `MetadataStateFormatTests` exercises this `catch`. Could you add a test that does?